lib/nm/index.ts

```diff
@@ -0,0 +1,377 @@
+		if (requestScanErr) {
+			throw formatError(500)(`Could not requestScan`)(requestScanErr);
+		}
+		const [getDeviceAccessPointsErr, {AccessPoints}] = await to(this.getObjectProperty(['org.freedesktop.NetworkManager.Device.Wireless','AccessPoints'])(this.devices.wifi.path));
+		if (getDeviceAccessPointsErr) {
+			throw formatError(500)(`Could not get AccessPoints property on ${this.devices.wifi.path}`)(getDeviceAccessPointsErr);
+		}
+		const getApProperties = () => _.map(AccessPoints, async (path) => {
+			const [getPropertiesErr, props] = await to(Bluebird.all([
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Ssid'])(path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Flags'])(path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'WpaFlags'])(path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'RsnFlags'])(path),
+				this.getObjectProperty(['org.freedesktop.NetworkManager.AccessPoint', 'Frequency'])(path),
+			]));
+			if (getPropertiesErr) {
+				throw formatError(500)(`Could not get ap ${path} property`)(getPropertiesErr);
+			}
+			return props;
+		});
+		const [getApPropertiesErr, rawAccessPoints] = await to(Bluebird.all(getApProperties()));
+		if (getApPropertiesErr) {
+			throw formatError(500)(`Could not getApProperties`)(getApPropertiesErr);
+		}
+		// Normalize AP data from array of arrays to single object for each AccessPoint
+		const accessPoints = _.map(rawAccessPoints, (rawProps: any[]) => {
+			const props = _.reduce(rawProps, (acc, prop: any) => {
+				return {
+					...acc,
+					...prop,
+				};
+			}, {});
+			return props;
+		});
+		this.accessPoints = accessPoints;
+		return makeNetworksReadable(this.accessPoints);
+	}
+
+	getCurrentNetwork = async () => {
+		const [getActiveConnectionsErr, [key, [connections]]] = await to(this.getActiveConnections());
+		if (getActiveConnectionsErr) {
+			throw formatError(500)(`Could not getActiveConnections`)(getActiveConnectionsErr);
+		}
+		const getConnectionsType = () => _.map(connections, this.getObjectProperty(['org.freedesktop.NetworkManager.Connection.Active', 'Type']));
+		const [getConnectionsTypeErr, results] = await to(Bluebird.all(getConnectionsType()));
+		if (getConnectionsTypeErr) {
+			throw formatError(500)(`Could not getConnectionsType`)(getConnectionsTypeErr);
+		}
+
+		const wifiConnection = _.filter(results, ({Type}) => Type === '802-11-wireless')[0];
+		const [getIdPropertyErr, {Id}] = await to(this.getObjectProperty(['org.freedesktop.NetworkManager.Connection.Active', 'Id'])(wifiConnection.path));
+		if (getIdPropertyErr) {
+			throw formatError(500)(`Could not get Id property on ${wifiConnection.path}`)(getIdPropertyErr);
+		}
+		return Id;
+	}
+}
+
+function formatError(type: number = 400) {
+	return (message: string) => (err: any = {}) => {
+		return {type, message, err};
+	};
+}
+
+function findConnection(connections) {
+	return (network) => {
```

Comment:
Instead of creating a function every time, might `findConnection(network, connections)` work as well?

Reply:
Yep, absolutely